lib/ace/ext/papeeria/papeeria_latex_highlight_rules.coffee

```diff
@@ -36,12 +36,19 @@ define((require, exports, module) ->
 
     popState = (currentState, stack) ->
       if not stack?
-        throw new Error("papeeria highlight rules stack error: stack doesn't exist")
+        throw new Error("papeeria_highlight_rules -- stack error: stack doesn't exist")
+
       if not Array.isArray(stack)
         return "start"
 
-      if not (currentState == "start" and stack.length == 0) && currentState != stack[stack.length - 1]
-        throw new Error("papeeria highlight rules stack error, expected " + stack[stack.length - 1] + " found " + currentState)
+      if stack.length == 0
+        if currentState != "start"
+         throw new Error('papeeria_highlight_rules -- stack error: stack should not be empty here')
+        else return "start"
+
+      # here we know stack is not empty
+      errorMessage = "papeeria_highlight_rules -- stack error: expected " + stack[stack.length - 1] + " found " + currentState
```

Comment:
is it really true? I thought that `currentState` is the external view of the state and what is actually expected, while the topmost element is what you have found